doc/user-getting-started-rust.md

```diff
@@ -0,0 +1,148 @@
+# Getting started with a native Rust program
+
+Please see [README](../README.md) for how to install Mystikos or build
+it from source code.
+
+Due to an issue in Rust's standard library, it is not currently possible to write Rust programs that use the standard library. See the end of this guide for more information on this issue.
+
+This guide assumes you have Rust installed through [`rustup`](https://rustup.rs) and that you're running on a Linux machine.
+
+You'll need to install the `x86_64-unknown-linux-musl` target which you can by running:
+
+```bash
+rustup target add x86_64-unknown-linux-musl
+```
+
+## Write the program (no_std)
+
+We'll write a fairly normal `no_std` Rust program. If you've not done `no_std` programming in Rust before, don't worry. While it's not quite as convenient as using the standard library, it's still fairly straight forward.
+
+```bash
+cargo new myapp
+cd myapp
+```
+
+Inside of `src/main.rs` paste the following code:
+
+```rust 
+/// We add a few annotations letting Rust know that we're making 
+#![no_std] // No standard library
+#![no_main] // We'll define our own main symbol
+
+#[no_mangle] // Make sure to expose this function as `main` and not some mangled name
+pub extern "C" fn main(_argc: isize, _argv: *const *const u8) -> isize {
+    unsafe {
+        // Print using `puts` from `libc`
+        libc::puts(b"Hello World!\0".as_ptr() as *const i8);
+    }
+    0
+}
+
+// We're required to specify a panic handler but we'll be aborting
+// on panic so no need to do anything here.
+#[panic_handler]
+fn panic(_info: &core::panic::PanicInfo) -> ! {
+        loop {}
+}
+
+// This is a workaround just for musl targets. By default in Rust, musl targets
+// statically link libc. This tells Rust to dynamically link libc.
+#[link(name = "c")]
+extern {}
+```
+
+We'll need to add a few things to our `Cargo.toml` manifest:
+
+```toml
+# Leave all the things in place up to `dependencies` which you replace with:
+[dependencies]
+libc = { version = "0.2", default-features = false } # bindings for libc
+
+[profile.release]
+panic = "abort" # abort on panic instead of unwinding
+lto = true # make our binary smaller and faster by performing link time optimizations
+```
+
+Lastly, we need to slightly change how Cargo builds our Rust project. We do this by making some changes in a Cargo config file which we create in our project directory at the path `.cargo/config`:
+
+```toml
+[target.'cfg(target_env = "musl")']
+rustflags = [
+    "-C", "linker=myst-gcc", # use `myst-gcc` for linking. `myst-gcc` is a wrapper for `musl-gcc` which links to MUSL libc instead of the default glibc.
```

Comment:
myst-gcc is not a complete replacement of gcc. I'd recommend to use regular gcc if possible. See my comment about glibc support below.